### ISSUE TITLE
No audio output

### ISSUE THREAD

**Issue by fedebruni84** [jmshrv/finamp#22]
Hi, I’m using Finamp on Android 7 on a DAP.
Songs start playing but no audio output on headphones.
I’ve also installed Jellyfin app and it’s working as expected.
Thanks

**Comment by UnicornsOnLSD** [comment]
Do you have any Audio Service logs?

**Comment by fedebruni84** [comment]
Yes,
seems that the problem is related to FLAC direct play. I have no problems with mp3 files. 

![Screenshot_20210413-124232](https://user-images.githubusercontent.com/37954254/114541369-ef3ffa00-9c56-11eb-859e-246e12cb677e.png)

**Comment by UnicornsOnLSD** [comment]
@bambadon @fedebruni84 This is a long shot, but could you see what codecs your devices support? This can be checked with https://play.google.com/store/apps/details?id=com.parseus.codecinfo

**Comment by vincegre** [comment]
Wanted to share my audio logs as requested but unhappy the copy button in logs page in Finamp is not working on my device :( it copies nothing in clipboard ! What can I do ?

**Comment by UnicornsOnLSD** [comment]
You could try downloading an album, switching to offline mode, and restarting the app. Your device is probably being overwhelmed by having to copy half of your Jellyfin library since the logs also include all communication with the server

**Comment by vincegre** [comment]
> You could try downloading an album, switching to offline mode, and restarting the app. Your device is probably being overwhelmed by having to copy half of your Jellyfin library since the logs also include all communication with the server

Not sure to understand as I have not setup any synchro so all medias are listened in streaming and side note my device is working perfectly well while Finamp running so I don't think it's overloaded ;)

**Comment by UnicornsOnLSD** [comment]
Any request is logged, so there will be a big log item with like 100 items in. Finamp can handle this fine, it's more the phone clipboard that can't handle it. Switching to offline mode will mean that no request is made, but you'll have to have something downloaded to actually test playback.

**Comment by vincegre** [comment]
> Any request is logged, so there will be a big log item with like 100 items in. Finamp can handle this fine, it's more the phone clipboard that can't handle it. Switching to offline mode will mean that no request is made, but you'll have to have something downloaded to actually test playback.

Thanks for explanation so synced few tracks, passed offline, restarted app and then copy/paste work, here it is below but it's not much verbose (couldnt find any options to get more verbose ones !)

```
[MusicPlayerBackgroundTask/INFO] 2021-09-15 18:26:04.740390: Starting audio service

null

[MusicPlayerBackgroundTask/INFO] 2021-09-15 18:26:04.740390: Starting audio service

null
```

**Comment by UnicornsOnLSD** [comment]
Did you actually play anything before copying those logs?

**Comment by fedebruni84** [comment]
> @bambadon @fedebruni84 This is a long shot, but could you see what codecs your devices support? This can be checked with https://play.google.com/store/apps/details?id=com.parseus.codecinfo

Codec list:

audio/3gpp (OMX.google.amrnb.decoder)
audio/3gpp (OMX.google.amrnb.encoder)
audio/amr-wb (OMX.google.amrwb.decoder)
audio/amr-wb (OMX.google.amrwb.encoder)
audio/flac (OMX.google.flac.encoder)
audio/g711-alaw (OMX.google.g711.alaw.decoder)
audio/g711-mlaw (OMX.google.g711.mlaw.decoder)
audio/gsm (OMX.google.gsm.decoder)
audio/mp4a-latm (OMX.google.aac.decoder)
audio/mp4a-latm (OMX.google.aac.encoder)
audio/mpeg (OMX.google.mp3.decoder)
audio/opus (OMX.google.opus.decoder)
audio/raw (OMX.google.raw.decoder)
audio/vorbis (OMX.google.vorbis.decoder)

It seems there is only flac encoder but not decoder??

**Comment by UnicornsOnLSD** [comment]
Hmm, looks like ExoPlayer has an extension for FLAC (https://github.com/google/ExoPlayer/tree/release-v2/extensions/flac). I'll look into using it.

**Comment by vincegre** [comment]
> Did you actually play anything before copying those logs?

Yep I played two tracks (that gave no sound for sure but app displayed them as playing ;) anything else I can do to help debug problem ?

**Comment by UnicornsOnLSD** [comment]
ExoPlayer has added a Java FLAC extractor for devices without native FLAC support (https://github.com/google/ExoPlayer/issues/6406). Finamp should be able to use this now.

**Comment by UnicornsOnLSD** [comment]
It should be in now, I don't think I have an easy way to test it though

**Comment by vincegre** [comment]
> It should be in now, I don't think I have an easy way to test it though

Sorry for the delay but problem has not changed, still no audio at all on my device :-1:

**Comment by UnicornsOnLSD** [comment]
Thanks for letting me know, this should be fixed once #177 is done

**Comment by Chaphasilor** [comment]
@vincegre is this still happening on your device? :)

**Comment by vincegre** [comment]
@Chaphasilor sorry but I don't use Finamp anymore since a while as I switched to Emby ;)

**Comment by Chaphasilor** [comment]
Closing this in favor of #177
